backends/gstreamer/src/player.rs

```diff
@@ -21,20 +23,48 @@ use std::u64;
 
 const MAX_BUFFER_SIZE: i32 = 500 * 1024 * 1024;
 
+struct GStreamerBuffer {
+    is_gl: bool,
+    frame: gst_video::VideoFrame<gst_video::video_frame::Readable>,
+}
+
+impl Buffer for GStreamerBuffer {
+    fn to_vec(&self) -> Result<FrameData, ()> {
+        // XXX: we support only one plane on one view
```

Comment:
You might want to ensure that you get BGRA or BGR or similar before the sink then. The packed formats are guaranteed to be in a single plane.

And you probably only support a specific format in the end anyway, probably some kind of RGB format?